src/jit/lowerxarch.cpp

```diff
@@ -921,11 +921,15 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     //         cmp dl, byte ptr[addr mode]
     //         movzx edx, dl
     //
+    // Example3: GT_ADD(int, op1 mem of byte sized value, op2 of type int) - add a byte sized value
+    // from mem and an int and store in mem op1. We need to make sure op2 is byteable.
+    //
     // Though this looks conservative in theory, in practice we could not think of a case where
     // the below logic leads to conservative register specification.  In future when or if we find
     // one such case, this logic needs to be fine tuned for that case(s).
     if (varTypeIsByte(tree) || ((tree->OperGet() == GT_CAST) && varTypeIsByte(tree->CastToType())) ||
-        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())))
+        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())) ||
+        ((tree->OperGet() == GT_ADD) && varTypeIsByte(tree->gtGetOp1()) && (tree->gtGetOp1()->OperGet() == GT_IND)))
```

Comment:
We need a more holistic fix than this.  Think of GT_SUB/GT_MUL node with similar IR.

I think we need to revisit how we can correctly set srcCandidates on operands confined to bytable regs where required.  

We need to examine to see if we can come up with a set of conditions that would characterize all the cases where bytable regs are required.  If it is not possible to come up with such a characterization, another option to examine is whether each GT_* node should set srcCandidates individually.